src/litDataset.ts

```diff
@@ -115,10 +115,72 @@ export async function fetchResourceInfo(
 
   const resourceInfo = parseResourceInfo(response);
 
-  return resourceInfo;
+  return { resourceInfo };
```

Comment:
Heh. I kinda expected this function to stay as-is, but I guess this does make sense given our earlier conversation. However, shouldn't `fetchAcl` then also return `unstable_WithAcl` rather than `unstable_WithAcl['acl']`?

(This is also a breaking change, but I assume Pod Manager isn't using this function yet.)

Reply:
I'm still quite hesitant. I'm really not a fan of `const resourceInfo = (await fetchResourceInfo).resourceInfo`, which the current proposal warrants. However, since all the functions of our interface expect a `WithAcl` rather than a `Acl`, `unstable_fetchResourceInfoWithAcl` returns a `WithResourceInfo & unstable_WithAcl`, and then for consistency it seems better than `unstable_fetchResourceInfo` returns a `WithResourceInfo`.

Reply:
But good point about `fetchAcl`: if it ever is no longer an internal function, it should be kept consistent with the API.

Reply:
Fixed in 89cb668f07cd0f0f8beb2d1fbbce91f3904f9279

Reply:
Yeah, I am more hesitant about `fetchResourceInfo` return a `WithResourceInfo` as well for some reason, though I'm finding it hard to justify why with actual arguments rather than a feeling of unease. (Especially since we're wrapping it in `{ resourceInfo }` in the only place we're using it anyway.)

I am wondering though: is `fetchResourceInfo` even relevant to consumers? Should we not mark it internal just like `fetchAcl`? Then we could also keep it as `WithResourceInfo['resourceInfo']` for now, until we can actually justify our feelings :P

Reply:
Well right now `fetchResourceInfo` doesn't do much, but are we sure we only want the API to expose the function `fetchResourceInfoAndAcl`, and not allowing lower-level `HEAD` requests that does not involve firing the additional `ACL` request ?

Reply:
Actually I can't think of an use-case that:
- Requires not to fetch the resource **AND**
- Requires to fetch the ACL link without fetching the ACL

Potentially some corner use-cases adding custom `Link` headers, but overall it seems fair to keep the API slim and focus on the use cases that have been expressed by actual users, namely "get the ACL without getting the resource". I'll make `fetchResourceInfo` internal for the time being.

Reply:
Exactly. Plus, exposing something later is easier than un-exposing it :)

Reply:
Fixed in 00c31bdae5f8959f34e0c89a355377d6506e30df